### PR TITLE
385 shell script options

### PR DESCRIPTION
**Pull request by JimHofman**
## Description
- Added --no-metrics and --no-cleanup switches to "test_int_<pge>.sh" shell scripts

## Affected Issues
- N/A
## Testing
- Tried out new switches and verified metrics were not generated and test files were not deleted

### CODE REVIEW

File: .ci/scripts/cslc_s1/test_int_cslc_s1.sh [hunk 1]
```diff
@@ -31,6 +31,11 @@ SAMPLE_TIME=15
 [ -z "${RUNCONFIG}" ] && RUNCONFIG="opera_pge_cslc_s1_delivery_6.1_final_runconfig.yaml"
 [ -z "${TMP_ROOT}" ] && TMP_ROOT="$DEFAULT_TMP_ROOT"
 
+echo "***********************"
```
[the remark this answers is not in the document]
Comment (JimHofman) [reply .ci/scripts/cslc_s1/test_int_cslc_s1.sh:34]: Deleted debug code

File: .ci/scripts/util/test_int_util.sh [hunk 2]
```diff
@@ -43,6 +47,15 @@ test_int_parse_args()
         shift
         shift
         ;;
+      --no-cleanup)
+        echo "Setting DELETE_TEMP_FILES to false. Temp files will not be deleted"
```
[the remark this answers is not in the document]
Comment (JimHofman) [reply .ci/scripts/util/test_int_util.sh:51]: deleted echo statement

File: .ci/scripts/util/test_int_util.sh [hunk 3]
```diff
@@ -147,22 +160,34 @@ test_int_trap_cleanup()
...
-        fi
-        rm "${stats_pid_file}"
+            process_to_kill=$(cat "${stats_pid_file}")
+            if ps -p $process_to_kill > /dev/null
+            then
+                kill $process_to_kill
+                wait $process_to_kill 2> /dev/null || true
+            fi
+            rm "${stats_pid_file}"
+      fi
```
[the remark this answers is not in the document]
Comment (JimHofman) [reply .ci/scripts/util/test_int_util.sh:189]: Aligned fi statement

File: .ci/scripts/util/util.sh [hunk 4]
```diff
@@ -121,71 +121,76 @@ copy_pge_files() {
...
-
-    dus="eval $block_space_cmd"
-    swu="eval $swap_space_cmd"
-    ths="eval $sys_threads_cmd"
-
-    { while true; do sleep "$sample_time"; \
-      echo "$(metrics_seconds)", "`$ds`", "`$dus`", "`$swu`", "`$ths`" >> "${metrics_stats}"; done } & \
-    echo "$!" > "${stats_pid_file}"
+  if $COLLECT_METRICS; then
+      echo "State of COLLECT_METRICS": $COLLECT_METRICS
```
[the remark this answers is not in the document]
Comment (JimHofman) [reply .ci/scripts/util/util.sh:125]: Removed debug statement.

File: .ci/scripts/util/test_int_util.sh [hunk 5]
```diff
@@ -147,22 +160,34 @@ test_int_trap_cleanup()
...
+        ${DOCKER_RUN} -v ${TMP_DIR}:${TMP_DIR} --entrypoint /usr/bin/find ${PGE_IMAGE}:${PGE_TAG} ${TMP_DIR} -type d -exec chmod 775 {} +
+        ${DOCKER_RUN} -v ${TMP_DIR}:${TMP_DIR} --entrypoint /usr/bin/find ${PGE_IMAGE}:${PGE_TAG} ${TMP_DIR} -type f -exec chmod 664 {} +
+        cd ${TMP_ROOT}
+        rm -rf ${TMP_DIR}
+    else
+        echo "--no-cleanup flag set: Temporary directories will remain on disk."
+    fi
+}
+
+test_end_metrics_collection()
```
[the remark this answers is not in the document]
Comment (JimHofman) [reply .ci/scripts/util/test_int_util.sh:175]: Yes, makes the old name is misleading.  Changed.

[the remark this answers is not in the document]
Comment (JimHofman) [reply .ci/scripts/util/test_int_util.sh:175]: Added function to test_int_util.sh that calls both the trap functions.  Used the name 'test_int_trap_cleanup so individual test files do not need to be updated.

File: .ci/scripts/util/test_int_util.sh [hunk 6]
```diff
@@ -186,8 +185,15 @@ test_end_metrics_collection()
...
-      fi
+        fi
      else
-         echo "--no-metrics flag set: Metrics collection is suspended. No need to stop collection."
+         echo "--no-metrics flag set"
     fi
 }
+
+# Clean up functions that are called on EXIT from test_int_<PGE>.sh scripts
+test_int_trap_cleanup()
```
Comment (collinss-jpl) [inline .ci/scripts/util/test_int_util.sh:195]: Clever!

**Review by collinss-jpl — APPROVED** [review]
Everything looks good @JimHofman, I am going to see if I can reproduce the qt-related error you encountered with this branch before merging

**Pull request merged**